File: 0114-flatten-binary-tree-to-linked-list/0114-flatten-binary-tree-to-linked-list.py
class Solution(object):
    def flatten(self, root):
        self.prev = None
        def dfs(node):
            if not node:
                return
            
            dfs(node.right)
            dfs(node.left)

            node.right = self.prev
            node.left = None
            self.prev = node
        dfs(root)            

File: 0114-flatten-binary-tree-to-linked-list/0114-flatten-binary-tree-to-linked-list.java
class Solution {
    public void flatten(TreeNode root) {
        if (root == null) return;

        flatten(root.left);
        flatten(root.right);

        TreeNode left = root.left;
        TreeNode right = root.right;

        root.left = null;
        root.right = left;

        TreeNode cur = root;
        while (cur.right != null) {
            cur = cur.right;
        }

        cur.right = right;
    }
}